jenkins/jobs/vic-scenario/build.sh

```diff
@@ -0,0 +1,149 @@
+export RUN_AS_OPS_USER=${RUN_AS_OPS_USER:-${DEFAULT_RUN_AS_OPS_USER}}
+
+PARALLEL_JOBS=${PARALLEL_JOBS:-${DEFAULT_PARALLEL_JOBS}}
+
+SCRIPT_DIR=$(cd $(dirname "$0") && pwd)
+
+if [[ $1 != "6.0" && $1 != "6.5" && $1 != "6.7" ]]; then
+    echo "Please specify a target version. One of: 6.0, 6.5, 6.7"
+    exit 1
+fi
+
+# process the CLI arguments
+target="$1"
+echo "Target version: ${target}"
+shift
+# Take the remaining CLI arguments as a test case list - this is treated as an array to preserve quoting when passing to pabot
+testcases=("${@:-${DEFAULT_TESTCASES[@]}}")
+
+# TODO: the version downloaded by this logic is not coupled with the tests that will be run against it. This should be altered to pull a version that matches the commit SHA of the tests
+# we will be running or similar mechanism.
+VCH_BUILD=${VCH_BUILD:-${DEFAULT_VCH_BUILD}}
+VCH_BRANCH=${VCH_BRANCH:-${DEFAULT_VCH_BRANCH}}
+ARTIFACT_BUCKET=${ARTIFACT_BUCKET:-${DEFAULT_ARTIFACT_BUCKET}}
+if [ "${VCH_BRANCH}" == "master" ]; then
+    GS_PATH="${ARTIFACT_BUCKET}"
+else
+    GS_PATH="${ARTIFACT_BUCKET}/${VCH_BRANCH}"
+fi
+input=$(gsutil ls -l gs://${GS_PATH}/${VIC_BINARY_PREFIX}${VCH_BUILD}.tar.gz | grep -v TOTAL | sort -k2 -r | head -n1 | xargs | cut -d ' ' -f 3 | xargs basename)
+constructed_url="https://storage.googleapis.com/${GS_PATH}/${input}"
+ARTIFACT_URL="${ARTIFACT_URL:-${constructed_url}}"
+input=$(basename ${ARTIFACT_URL})
+
+# strip prefix and suffix from archive filename
+VCH_BUILD=${input#${VIC_BINARY_PREFIX}}
+VCH_BUILD=${VCH_BUILD%%.*}
+
+# Enforce short SHA
+GIT_COMMIT=${GIT_COMMIT:0:7}
+
+case "$target" in
+    "6.0")
+        excludes="--exclude nsx"
+        ESX_BUILD=${ESX_BUILD:-$ESX_60_VERSION}
+        VC_BUILD=${VC_BUILD:-$VC_60_VERSION}
+        ;;
+    "6.5")
+        ESX_BUILD=${ESX_BUILD:-$ESX_65_VERSION}
+        VC_BUILD=${VC_BUILD:-$VC_65_VERSION}
+        ;;
+    "6.7")
+        excludes="--exclude nsx --exclude hetero"
+        ESX_BUILD=${ESX_BUILD:-$ESX_67_VERSION}
+        VC_BUILD=${VC_BUILD:-$VC_67_VERSION}
+        ;;
+esac
+
+LOG_UPLOAD_DEST="${LOG_UPLOAD_DEST:-${DEFAULT_LOG_UPLOAD_DEST}}"
+
+pushd vic
+    n=0 && rm -f "${input}"
+    until [ $n -ge 5 -o -f "${input}" ]; do
+        echo "Retry.. $n"
+        echo "Downloading gcp file ${input} from ${ARTIFACT_URL}"
+        wget -nv ${ARTIFACT_URL}
+
+        ((n++))
+        sleep 15
+    done
+
+    echo "Extracting .tar.gz"
+    mkdir bin && tar xvzf ${input} -C bin/ --strip 1
+
+    if [ ! -f  "bin/vic-machine-linux" ]; then
+        echo "Tarball extraction failed..quitting the run"
+        rm -rf bin
+        exit
+    else
+        VCH_COMMIT=$(bin/vic-machine-linux version | awk -F '-' '{print $NF}')
+        echo "Tarball extraction passed, Running nightlies test.."
+    fi
+
+    pabot --processes ${PARALLEL_JOBS} --removekeywords TAG:secret ${excludes} --variable ESX_VERSION:${ESX_BUILD} --variable VC_VERSION:${VC_BUILD} -d report "${testcases[@]}"
```

Comment:
quote the ESX_BUILD and VC_BUILD to match the vic-product usage

Reply:
OK